### ISSUE TITLE
The callback error does not work and an exception error is generated!

### ISSUE THREAD

**Issue by NabiKAZ** [kriasoft/node-sqlite#160]
This sample code is exactly the [same](https://github.com/kriasoft/node-sqlite#each) as the sample code on the first page of the project:

```js
await db2.each(
	'SELECT * FROM test WHERE score1 = ?', [1001],
	(err, row) => {
		if (err) {
			console.log("err", err.message);
		}
		console.log("row", row);
	}
)
```

Here for error handling, a callback is used in the third parameter of the function.
But when an error occurs, this callback is ignored and an exception error is generated:

```js
node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^

[Error: SQLITE_ERROR: no such column: score1] {
  errno: 1,
  code: 'SQLITE_ERROR'
}

Node.js v17.2.0
```

**And the program stops!**
I had to use try/catch to manage it:

```js
try {
	await db2.each(
		'SELECT * FROM test WHERE score1 = ?', [1001],
		(err, row) => {
			console.log("row", row);
		}
	)
} catch(err) {
	console.log("err", err.message);
}
```

That works well:

```js
err SQLITE_ERROR: no such column: score1
```

However, that callback works well in the sqlite3 parent project and does not generate an exception error.
Basically, if this is not a bug, then why is this method mentioned in the guide, and if it is a bug, I think it can be solved.

**Comment by theogravity** [comment]
I looked into this and there's actually two kinds of errors that can be thrown. 

- At the function call level. This could be SQL parse or connection errors.
- At the callback level. It's not clear what kind of errors are thrown at the callback level.

The implementation properly follows the documented API:

https://github.com/TryGhost/node-sqlite3/wiki/API#databaseeachsql-param--callback-complete

I have updated the readme to clarify this further.

**Comment by NabiKAZ** [comment]
You said the implementation follows the original documentation.
But in this example you can see that exactly the same code calls the callback by the original project (db) but by this project (db2) an exception error is generated.

```js
import sqlite3 from 'sqlite3';
var db = new sqlite3.Database('test.db');

import { open } from 'sqlite'
const db2 = await open({
  filename: 'test.db',
  driver: sqlite3.Database
});
```

#### db -> sqlite3 [original]
```js
await db.each(
	'SELECT * FROM test WHERE score1 = ?', [1001],
	(err, row) => {
		console.log(">>> ERR:", err);
		console.log("row", row);
	}
);
```
result:
```js
>>> ERR: [Error: SQLITE_ERROR: no such column: score1] {
  errno: 1,
  code: 'SQLITE_ERROR'
}
row undefined
```

#### db2 -> sqlite [this]
```js
await db2.each(
	'SELECT * FROM test WHERE score1 = ?', [1001],
	(err, row) => {
		console.log(">>> ERR:", err);
		console.log("row", row);
	}
);
```
result:
```js
node:internal/process/esm_loader:94
    internalBinding('errors').triggerUncaughtException(
                              ^

[Error: SQLITE_ERROR: no such column: score1] {
  errno: 1,
  code: 'SQLITE_ERROR'
}

Node.js v17.2.0
```

**Comment by theogravity** [comment]
Here's the implementation:

```
      this.db.each(
        sqlObj.sql,
        ...sqlObj.params,
        (err, row) => {
          if (err) {
            return callback(formatError(err), null)
          }

          callback(null, row)
        },
        (err, count) => {
          if (err) {
            return reject(formatError(err))
          }

          resolve(count)
        }
      )
```

If there is an error in the `[callback]`, it's not going to reject the promise, but instead call the callback with the error populated. Your situation happens because it's a direct error, and so the error is thrown in the `[complete]` callback.

If you think this is incorrect, please put up a PR + tests with what you think is the right solution and it will be reviewed.

**Comment by NabiKAZ** [comment]
I did not understand exactly what happened! But it does not matter! (:
My only goal was to see the difference in functionality between your version and the original version, which you seem to understand, and which seems normal.
Thanks.
